core/trino-parser/src/main/java/io/trino/sql/parser/AstBuilder.java

```diff
@@ -2042,6 +2042,27 @@ public Node visitExtract(SqlBaseParser.ExtractContext context)
         return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), field);
     }
 
+    @Override
+    public Node visitTrim(SqlBaseParser.TrimContext context)
+    {
+        String functionName = getTrimFunctionName((Token) context.trimsSpecification().getChild(0).getPayload());
+        List<Expression> arguments = Lists.reverse(visit(context.valueExpression(), Expression.class));
+        return new FunctionCall(getLocation(context), QualifiedName.of(functionName), arguments);
+    }
```

Comment:
Ideally, this should create a Trim AST node. The mapping to a function should be done during query planning.

In particular, the parsing operation should be reversible -- it should be possible to generate syntactically equivalent SQL from the parsed syntax tree (minus formatting)

Reply:
I'm rewriting to return Trim AST node, but it might be little complicated. New trim function takes two string arguments, but existing function takes a code points type as the 2nd argument. If we change the node `trim|ltrim|rtrim` functions in AstBuilder.visitFunctionCall to `Trim` from `FunctionCall`, it leads to test failure in `TreeAssertions.assertListEquals`. If we don't change the node in visitFunctionCall, it throws an exception when invoking the function in ExpressionInterpreter.

Reply:
If all `TRIM` are parsed by the same rule, and represented by the same AST node, that might solve this issue.